content/docs/tasks/traffic-management/debugging-istio/index.md

```diff
@@ -0,0 +1,187 @@
+reviews-v2-686bbb668-99j76.default                     SYNCED     SYNCED     SYNCED (100%)     SYNCED       istio-pilot-75bdf98789-tfdvh
+reviews-v3-7b9b5fdfd6-4r52s.default                    SYNCED     SYNCED     SYNCED (100%)     SYNCED       istio-pilot-75bdf98789-n2kqh
+{{< /text >}}
+
+If a proxy is missing from this list it means that it is not currently connected to a Pilot instance so will not be receiving any configuration.
+
+* `SYNCED` means that Envoy has acknowledged the last configuration Pilot has sent to it.
+* `SYNCED (100%)` means that Envoy has successfully synced all of the endpoints in the cluster.
+* `NOT SENT` means that Pilot hasn't sent anything to Envoy. This usually is because Pilot has nothing to send.
+* `STALE` means that Pilot has sent an update to Envoy but has not received an acknowledgement. This usually indicates a networking issue between Envoy and Pilot or a bug with Istio itself.
+
+## Retrieve diffs between Envoy and Pilot
+
+The `proxy-status` command can also be used to retrieve a diff between the configuration Envoy has loaded and the configuration Pilot would send, by providing a proxy ID. This can help you determine exactly what is out of sync and where the issue may lie.
+
+{{< text bash >}}
+$ istioctl proxy-status details-v1-6dcc6fbb9d-wsjz4.default
+--- Pilot Clusters
++++ Envoy Clusters
+@@ -374,36 +374,14 @@
+             "edsClusterConfig": {
+                "edsConfig": {
+                   "ads": {
+
+                   }
+                },
+                "serviceName": "outbound|443||public-cr0bdc785ce3f14722918080a97e1f26be-alb1.kube-system.svc.cluster.local"
+-            },
+-            "connectTimeout": "1.000s",
+-            "circuitBreakers": {
+-               "thresholds": [
+-                  {
+-
+-                  }
+-               ]
+-            }
+-         }
+-      },
+-      {
+-         "cluster": {
+-            "name": "outbound|53||kube-dns.kube-system.svc.cluster.local",
+-            "type": "EDS",
+-            "edsClusterConfig": {
+-               "edsConfig": {
+-                  "ads": {
+-
+-                  }
+-               },
+-               "serviceName": "outbound|53||kube-dns.kube-system.svc.cluster.local"
+             },
+             "connectTimeout": "1.000s",
+             "circuitBreakers": {
+                "thresholds": [
+                   {
+
+                   }
+
+Listeners Match
+Routes Match
+{{< /text >}}
+
+Here we can see that the listeners and routes match but the clusters are out of sync.
+
+## Deep dive into Envoy configuration
+
+The `proxy-config` command can be used to see how a given Envoy instance is configured. This can then be used to pinpoint any issues you are unable to detect by just looking through your Istio configuration and custom resources. To get a basic summary of clusters, listeners or routes for a given pod use the command as follows (changing clusters for listeners or routes when required):
```

Comment:
Since you volunteered for talking about deep dive into Istio generated envoy config, why don't you add some more text on how Pilot generates listeners for a given sidecar? 
On sidecars:
One listener on port 15003 that receives traffic from IPtables, and hands over to a virtual listener
A virtual listener per service IP per each non-HTTP ports for outbound TCP/HTTPS traffic
A virtual listener on the pod IP for each endpoint port for inbound traffic
A virtual listener on 0.0.0.0 per each HTTP port for outbound HTTP traffic.

On gateways:
A listener for each unique port in the gateway configuration

Reply:
Was hoping you wouldn't pick on this and I would avoid going into the more complicated sidecar config. I'll change it now.

Isn't the IP tables listener on 15001?